### ISSUE TITLE
add action search usability regression: "set text"

### ISSUE THREAD

**Issue by F3der1co** [Scirra/Construct-bugs#7618]
## Problem description
Since the addition of set text direction searching "set text" (or "set t" etc.) highlights "set text direction" in the add action dialog. (Because it's higher up in the order I guess)
Set Text is such a common thing to use and search for, that it can be really annoying, I'm not sure how best to fix it, maybe renaming the "set text direction action" so it doesn't lead to that issue.

## Attach a .c3p
[setText.zip](https://github.com/Scirra/Construct-bugs/files/13560234/setText.zip)

## Steps to reproduce

1. open the add action dialogue
2. start typing "set text"

## Observed result
"set text direction" is highlighted

## Expected result
"set text" is highlighted

## More details

**Affected browsers/platforms:** 

**First affected release:** 

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by AshleyScirra** [comment]
Yeah, it's definitely better Construct highlights "Set text" in that case. I also found that if you search for "app", it highlights "Set wrapping" rather than "Append text", which seems the more likely intention.

I've updated the default result selection heuristics for the next beta with a slightly cleverer system that prefers items that start with the match, and also prefers shorter item texts (so "Set text" is preferred over "Set text direction"). This seems to fix both cases and works nicely from my own tests, but as it's more or less a guess it might get some other cases wrong or throw off people's existing habits for searches - we'll see how it works out over the beta cycle.
